txt_files/config.txt
import os

# --- Шляхи та папки ---
BASE_DIR = os.path.dirname(os.path.abspath(__file__))
INPUT_FOLDER = os.path.join(BASE_DIR, 'input_data')
OUTPUT_FOLDER = os.path.join(BASE_DIR, 'output')
ARCHIVE_FOLDER = os.path.join(INPUT_FOLDER, 'processed_archive')
OUTPUT_FILE = os.path.join(OUTPUT_FOLDER, 'Total_Ledger.xlsx')
LOG_FILE = os.path.join(BASE_DIR, 'automation.log')

# --- Режим роботи ---
REBUILD_MODE = False  # False — щоденний дозапис, True — повний перезапис

# --- Константи назв колонок ---
COL_ID = 'ID_Транзакції'
COL_DATE = 'Дата'
COL_CAT = 'Категорія'
COL_CARD = 'Картка'
COL_DESC = 'Опис операції'
COL_AMOUNT = 'Сума'
COL_BALANCE = 'Залишок'
COL_MCC = 'MCC'
COL_CLEARING_STATUS = 'Статус клірингу'

USEFUL_COLUMNS = [COL_DATE, COL_CAT, COL_CARD, COL_DESC, COL_AMOUNT, COL_BALANCE, COL_CLEARING_STATUS]

# --- Словник карток ---
CARDS_DICTIONARY = {
    '4169': 'Приват Sid Унів',
    '4882': 'Приват Sid Кешбек',
    '0707': 'Приват Sid Виплати',
    '0627': 'Приват Olenka Виплати',
    '6333': 'Приват Olenka Кешбек',
    '4258': 'Приват Olenka Унів',
    '9948': 'А-Банк Sid АТБ',
    '0430': 'А-Банк Sid АТБ',
    '4907': 'А-Банк Sid Спорт',
    '0956': 'А-Банк Sid Кешбек',
    '5191': 'А-Банк Sid Кешбек',
    '5993': 'Альянс Sid',
    '0550': 'Альянс Olenka',
    '4805': 'ПУМБ Sid',
    '8169': 'ПУМБ Sid кредит',
    '0950': 'ПУМБ Olenka',
    '7854': 'Monobank Sid Чорна',
}

IBAN_TO_CARD_MAP = {
    'UA093052990262006400941962343': 'Приват Olenka Виплати',
    'UA173052990000026201664106959': 'Приват Olenka Унів',
    'UA693052990262076401003353719': 'Приват Olenka Кешбек',
    'UA703052990000026203685207802': 'Приват Sid Унів',
    'UA503052990262056401002870255': 'Приват Sid Кешбек',
    'UA093052990262006400928529686': 'Приват Sid Виплати',
    'UA823077700000026202412881084': 'А-Банк Sid АТБ',
    'UA703077700000026201117655044': 'А-Банк Sid Кешбек',
    'UA983077700000026207321065082': 'А-Банк Sid Спорт',
    'UA903001190000262071247640001': 'Альянс Sid',
    'UA373001190000262041247667001': 'Альянс Olenka',
    'UA773220010000026201300425413': 'Monobank Sid Чорна',
    'UA363348510000026203122871748': 'ПУМБ Sid',
    'UA233348510000026200123291900': 'ПУМБ Sid кредит',
    'UA913348510000026209122872688': 'ПУМБ Olenka',
}


MCC_MAP = {
    # --- Прихідні категорії ---
    'бонуси та кешбек': [],
    'зарплата і виплати': [],
    'поповнення готівкою': ['4829'],
    'переказ з власного рахунку': [],
    'переказ з чужого рахунку': ['6012'],
    'інші надходження': [],
    # --- Розхідні категорії ---
    'продукти': ['5411', '5499'],
    'ресторани та розваги': ['5814', '5812'],
    "здоров'я та краса": ['5912', '5995', '8011', '8099'],
    'витрати на авто': ['5541', '5542'],
    'громадський транспорт': ['4121', '4111', '4112'],
    'поштові послуги': ['4215', '9402'],
    'техніка і цифрові товари': ['5310', '5942', '5943'],
    'компослуги': ['4814', '4900'],
    'одяг та взуття': ['5719', '5651'],
    "зв'язок": ['4812'],
    'спорт': ['7941', '7997'],
    'зняття готівки': ['6011'],
    'оплата кредитів': [],
    'житло та побут': ['5211', '5251', '5261'],
    'переказ на власний рахунок': [],
    'переказ на чужий рахунок': [],
    'інші витрати': ['6010', '6300', '9399'],
}

### --- НОВИЙ ГРУПОВАНИЙ СЛОВНИК ---
CATEGORIES_KEYWORDS = {
    # --- Прихідні категорії ---
    'бонуси та кешбек': [
        'національний кешбек', 'abnk', 'зимова підтримка', 'кешбек', 'зарахування кешбек',
        'запроси друга', 'виплата винагороди', 'bonus', 'зарахування «національний кешбек»',
        'переказ на картку кешбек',
    ],
    'зарплата і виплати': [
        'zarplat', 'hroshove zabezpechennia', 'військова частина', 'вч а', 'міська рада',
        'соціальні виплати', 'матерiальна допомога', 'грошове забезпечення', 'зарплата', 'хзввсп',
    ],
    'поповнення готівкою': [
        'city24', 'city', 'термінал mono', 'термінал самообслуговування',
        'поповнення (city24)', 'термінал',
    ],
    'переказ з власного рахунку': [
        'зі своєї картки', 'зі своєї карти', 'із власної картки',
    ],
    'переказ з чужого рахунку': [
        'від:', 'від', 'vid:', 'paysend', 'kyiv ua', 'переказ на картку',
    ],
    'інші надходження': [
        'повернення коштів',
    ],
    # ---Розхідні категорії---
    'витрати на авто': [
        'укрнафта', 'wog', 'amic', 'azs', 'azs shop', 'azk', 'strakhovyi', 'avtomyika', 'doktor kolesa',
        'автостиль', 'автореал', 'авто', 'автомийка',
    ],
    'громадський транспорт': [
        'укрзалізниця', 'uz', 'vokzal', 'леокарт', 'kyivtrans', 'bolt', 'uklon', 'квиток',
        'металургів', 'е-квиток',
    ],
    'житло та побут': [
        'епіцентр', 'brazzers', 'господар', 'грядка', 'альфа плюс', 'eva', 'таке', 'posud',
        'астра', 'е-таке', 'магазин господар', 'ваша грядка',
    ],
    "зв'язок": [
        'vodafone', 'київстар', 'kyivstar', 'plus380', 'поповнення мобільного',
    ],
    "здоров'я та краса": [
        'makeup', 'дружба', 'vetapteka', 'medical', 'medychnyi', 'аптека', 'apteka',
        "men's hall", 'wfp consult', 'конекс', 'фарм', 'подорожник', 'wfp.medicalservices', 'makeup',
    ],
    'одяг та взуття': [
        'sinsay', 'bershka', 'deva', 'vzuttya', 'lileya', 'dzhins', 'магазин глобус', 'лпп', 'lpp',
        'economclass', 'planeta dzhins', 'магазин взуття',
    ],
    'зняття готівки': [
        'зняття готівки', 'видача готівки', 'зняття готівки в банкоматі',
    ],
    'компослуги': [
        'нафтогаз', 'газ', 'комунальні', 'електроенергію', 'ipay', 'portmone',
        'хмельницькенергозбут', 'шепетівкагаз', 'internet', 'реп', 'вивіз тпв',
    ],
    'оплата кредитів': [
        'кредит', 'погашення', 'заборгованість', 'позичальник',
    ],
    'поштові послуги': [
        'novapay', 'нова пошта', 'nova poshta', 'укрпошта', 'meest',
    ],
    'продукти': [
        'атб', 'atb', 'сільпо', 'сильпо', 'сімі', 'точка', 'окорська', 'березовський', 
        'гастроном', 'metro', 'траш', 'krai', 'kray', 'thrash', 'family', 'близенько',
        'pavliuk', 'велмарт', 'velmart', 'gwater', 'dobry pan', 'маркет', 'магазин',
        'shop', 'roshen', 'napoi', 'фермер', 'лаваш', 'продукти', 'taranenko', 'magazyn', 'арсен',
        'диканька', 'щодня', 'koshik', 'magazinkoshik', 'кошик', 'тайстра', 'м\'ясний майстер',
        'мясний маркет', 'fish place', 'rybka tvoey', 'фора',
    ],
    'ресторани та розваги': [
        'какао', 'coffee', 'монастiк', 'wings', 'пекарня', 'іграш', 'entirzhi', 'чудо',
        'vyishnya', 'ekspres', 'kafeteriy', 'кавярня', 'кафе', 'сушия', 'osama', 'кав`ярня',
        'kava', 'кава', 'mcdonald', 'kfc', 'pizza', 'sushi', 'кебаб', 'буфет', 'rybachok',
        'ewings', 'magic', 'shaurma', 'vape', 'vape land shop', 'vm shops', 'vm',
        'будинок іграшок', 'maisternia sho', 'aroma kava', 'pyvo', 'coffee bike',
    ],
    'спорт': [
        'спорт', 'vovkulaky', 'biont', 'спортивна школа',
    ],
    'техніка і цифрові товари': [
        'rozetka', 'aliexpress', 'temu', 'аврора', 'sklad magazyn', 'канцелярія', 'foxtrot',
        'фотохатка', 'kristal', 'comics', 'цифрові', 'amazon', "комп'ютерний всесвіт",
    ],
    'переказ на власний рахунок': [
        'на свою картку',
    ],
    'переказ на чужий рахунок': [
        'на картку', 'від'
    ],
    'інші витрати': [
        'комісія',
    ],
}

### --- СПИСКИ ДЛЯ ЛОГІКИ ---
INCOME_CATEGORIES = ['бонуси та кешбек', 'зарплата і виплати', 'інші надходження', 'переказ з власного рахунку', 'переказ з чужого рахунку', 'поповнення готівкою', 'транзит Віка']

EXPENSES_CATEGORIES = ['витрати на авто', 'громадський транспорт', 'житло та побут', "зв'язок", "здоров'я та краса", 'одяг та взуття', 'зняття готівки', 'компослуги', 'оплата кредитів', 'поштові послуги', 'продукти', 'ресторани та розваги', 'спорт', 'техніка і цифрові товари', 'переказ на власний рахунок', 'переказ на чужий рахунок', 'інші витрати', 'інвестиції']

AMBIGUOUS_CATEGORIES = ['переказ', 'оплата', 'інше', 'нерозпізнана', 'покупка', 'перекази']

EXCEL_MAPPING = {
    'дата': 'Дата',
    'категорія': 'Категорія',
    'картка': 'Картка',
    'опис операції': 'Опис операції',
    'сума в валюті картки': 'Сума',
    'залишок на кінець періоду': 'Залишок',
}


config.py
import os

# --- Шляхи та папки ---
BASE_DIR = os.path.dirname(os.path.abspath(__file__))
INPUT_FOLDER = os.path.join(BASE_DIR, 'input_data')
OUTPUT_FOLDER = os.path.join(BASE_DIR, 'output')
ARCHIVE_FOLDER = os.path.join(INPUT_FOLDER, 'processed_archive')
OUTPUT_FILE = os.path.join(OUTPUT_FOLDER, 'Total_Ledger.xlsx')
LOG_FILE = os.path.join(BASE_DIR, 'automation.log')

# --- Константи назв колонок ---
COL_ID = 'ID_Транзакції'
COL_DATE = 'Дата'
COL_CAT = 'Категорія'
COL_CARD = 'Картка'
COL_DESC = 'Опис операції'
COL_AMOUNT = 'Сума'
COL_BALANCE = 'Залишок'
COL_MCC = 'MCC'
COL_CLEARING_STATUS = 'Статус клірингу'

USEFUL_COLUMNS = [COL_DATE, COL_CAT, COL_CARD, COL_DESC, COL_AMOUNT, COL_BALANCE, COL_CLEARING_STATUS]

# --- Словник карток ---
CARDS_DICTIONARY = {
    '4169': 'Приват Sid Унів',
    '4882': 'Приват Sid Кешбек',
    '0707': 'Приват Sid Виплати',
    '0627': 'Приват Olenka Виплати',
    '6333': 'Приват Olenka Кешбек',
    '4258': 'Приват Olenka Унів',
    '9948': 'А-Банк Sid АТБ',
    '0430': 'А-Банк Sid АТБ',
    '4907': 'А-Банк Sid Спорт',
    '0956': 'А-Банк Sid Кешбек',
    '5191': 'А-Банк Sid Кешбек',
    '5993': 'Альянс Sid',
    '0550': 'Альянс Olenka',
    '4805': 'ПУМБ Sid',
    '8169': 'ПУМБ Sid кредит',
    '0950': 'ПУМБ Olenka',
    '7854': 'Monobank Sid Чорна',
}

IBAN_TO_CARD_MAP = {
    'UA093052990262006400941962343': 'Приват Olenka Виплати',
    'UA173052990000026201664106959': 'Приват Olenka Унів',
    'UA693052990262076401003353719': 'Приват Olenka Кешбек',
    'UA703052990000026203685207802': 'Приват Sid Унів',
    'UA503052990262056401002870255': 'Приват Sid Кешбек',
    'UA093052990262006400928529686': 'Приват Sid Виплати',
    'UA823077700000026202412881084': 'А-Банк Sid АТБ',
    'UA703077700000026201117655044': 'А-Банк Sid Кешбек',
    'UA983077700000026207321065082': 'А-Банк Sid Спорт',
    'UA903001190000262071247640001': 'Альянс Sid',
    'UA373001190000262041247667001': 'Альянс Olenka',
    'UA773220010000026201300425413': 'Monobank Sid Чорна',
    'UA363348510000026203122871748': 'ПУМБ Sid',
    'UA233348510000026200123291900': 'ПУМБ Sid кредит',
    'UA913348510000026209122872688': 'ПУМБ Olenka',
}


MCC_MAP = {
    # --- Прихідні категорії ---
    'бонуси та кешбек': [],
    'зарплата і виплати': [],
    'поповнення готівкою': ['4829'],
    'переказ з власного рахунку': [],
    'переказ з чужого рахунку': ['6012'],
    'інші надходження': [],
    # --- Розхідні категорії ---
    'продукти': ['5411', '5499'],
    'ресторани та розваги': ['5814', '5812'],
    "здоров'я та краса": ['5912', '5995', '8011', '8099'],
    'витрати на авто': ['5541', '5542'],
    'громадський транспорт': ['4121', '4111', '4112'],
    'поштові послуги': ['4215', '9402'],
    'техніка і цифрові товари': ['5310', '5942', '5943'],
    'компослуги': ['4814', '4900'],
    'одяг та взуття': ['5719', '5651'],
    "зв'язок": ['4812'],
    'спорт': ['7941', '7997'],
    'зняття готівки': ['6011'],
    'оплата кредитів': [],
    'житло та побут': ['5211', '5251', '5261'],
    'переказ на власний рахунок': [],
    'переказ на чужий рахунок': [],
    'інші витрати': ['6010', '6300', '9399'],
}

### --- НОВИЙ ГРУПОВАНИЙ СЛОВНИК ---
CATEGORIES_KEYWORDS = {
    # --- Прихідні категорії ---
    'бонуси та кешбек': [
        'національний кешбек', 'abnk', 'зимова підтримка', 'кешбек', 'зарахування кешбек',
        'запроси друга', 'виплата винагороди', 'bonus', 'зарахування «національний кешбек»',
        'переказ на картку кешбек',
    ],
    'зарплата і виплати': [
        'zarplat', 'hroshove zabezpechennia', 'військова частина', 'вч а', 'міська рада',
        'соціальні виплати', 'матерiальна допомога', 'грошове забезпечення', 'зарплата', 'хзввсп',
    ],
    'поповнення готівкою': [
        'city24', 'city', 'термінал mono', 'термінал самообслуговування',
        'поповнення (city24)', 'термінал',
    ],
    'переказ з власного рахунку': [
        'зі своєї картки', 'зі своєї карти', 'із власної картки',
    ],
    'переказ з чужого рахунку': [
        'від:', 'від', 'vid:', 'paysend', 'kyiv ua', 'переказ на картку',
    ],
    'інші надходження': [
        'повернення коштів',
    ],
    # ---Розхідні категорії---
    'витрати на авто': [
        'укрнафта', 'wog', 'amic', 'azs', 'azs shop', 'azk', 'strakhovyi', 'avtomyika', 'doktor kolesa',
        'автостиль', 'автореал', 'авто', 'автомийка',
    ],
    'громадський транспорт': [
        'укрзалізниця', 'uz', 'vokzal', 'леокарт', 'kyivtrans', 'bolt', 'uklon', 'квиток',
        'металургів', 'е-квиток',
    ],
    'житло та побут': [
        'епіцентр', 'brazzers', 'господар', 'грядка', 'альфа плюс', 'eva', 'таке', 'posud',
        'астра', 'е-таке', 'магазин господар', 'ваша грядка',
    ],
    "зв'язок": [
        'vodafone', 'київстар', 'kyivstar', 'plus380', 'поповнення мобільного',
    ],
    "здоров'я та краса": [
        'makeup', 'дружба', 'vetapteka', 'medical', 'medychnyi', 'аптека', 'apteka',
        "men's hall", 'wfp consult', 'конекс', 'фарм', 'подорожник', 'wfp.medicalservices', 'makeup',
    ],
    'одяг та взуття': [
        'sinsay', 'bershka', 'deva', 'vzuttya', 'lileya', 'dzhins', 'магазин глобус', 'лпп', 'lpp',
        'economclass', 'planeta dzhins', 'магазин взуття',
    ],
    'зняття готівки': [
        'зняття готівки', 'видача готівки', 'зняття готівки в банкоматі',
    ],
    'компослуги': [
        'нафтогаз', 'газ', 'комунальні', 'електроенергію', 'ipay', 'portmone',
        'хмельницькенергозбут', 'шепетівкагаз', 'internet', 'реп', 'вивіз тпв',
    ],
    'оплата кредитів': [
        'кредит', 'погашення', 'заборгованість', 'позичальник',
    ],
    'поштові послуги': [
        'novapay', 'нова пошта', 'nova poshta', 'укрпошта', 'meest',
    ],
    'продукти': [
        'атб', 'atb', 'сільпо', 'сильпо', 'сімі', 'точка', 'окорська', 'березовський', 
        'гастроном', 'metro', 'траш', 'krai', 'kray', 'thrash', 'family', 'близенько',
        'pavliuk', 'велмарт', 'velmart', 'gwater', 'dobry pan', 'маркет', 'магазин',
        'shop', 'roshen', 'napoi', 'фермер', 'лаваш', 'продукти', 'taranenko', 'magazyn', 'арсен',
        'диканька', 'щодня', 'koshik', 'magazinkoshik', 'кошик', 'тайстра', 'м\'ясний майстер',
        'мясний маркет', 'fish place', 'rybka tvoey', 'фора',
    ],
    'ресторани та розваги': [
        'какао', 'coffee', 'монастiк', 'wings', 'пекарня', 'іграш', 'entirzhi', 'чудо',
        'vyishnya', 'ekspres', 'kafeteriy', 'кавярня', 'кафе', 'сушия', 'osama', 'кав`ярня',
        'kava', 'кава', 'mcdonald', 'kfc', 'pizza', 'sushi', 'кебаб', 'буфет', 'rybachok',
        'ewings', 'magic', 'shaurma', 'vape', 'vape land shop', 'vm shops', 'vm',
        'будинок іграшок', 'maisternia sho', 'aroma kava', 'pyvo', 'coffee bike',
    ],
    'спорт': [
        'спорт', 'vovkulaky', 'biont', 'спортивна школа',
    ],
    'техніка і цифрові товари': [
        'rozetka', 'aliexpress', 'temu', 'аврора', 'sklad magazyn', 'канцелярія', 'foxtrot',
        'фотохатка', 'kristal', 'comics', 'цифрові', 'amazon', "комп'ютерний всесвіт",
    ],
    'переказ на власний рахунок': [
        'на свою картку',
    ],
    'переказ на чужий рахунок': [
        'на картку', 'від'
    ],
    'інші витрати': [
        'комісія',
    ],
}

### --- СПИСКИ ДЛЯ ЛОГІКИ ---
INCOME_CATEGORIES = ['бонуси та кешбек', 'зарплата і виплати', 'інші надходження', 'переказ з власного рахунку', 'переказ з чужого рахунку', 'поповнення готівкою', 'транзит Віка']

EXPENSES_CATEGORIES = ['витрати на авто', 'громадський транспорт', 'житло та побут', "зв'язок", "здоров'я та краса", 'одяг та взуття', 'зняття готівки', 'компослуги', 'оплата кредитів', 'поштові послуги', 'продукти', 'ресторани та розваги', 'спорт', 'техніка і цифрові товари', 'переказ на власний рахунок', 'переказ на чужий рахунок', 'інші витрати', 'інвестиції']

AMBIGUOUS_CATEGORIES = ['переказ', 'оплата', 'інше', 'нерозпізнана', 'покупка', 'перекази']

EXCEL_MAPPING = {
    'дата': 'Дата',
    'категорія': 'Категорія',
    'картка': 'Картка',
    'опис операції': 'Опис операції',
    'сума в валюті картки': 'Сума',
    'залишок на кінець періоду': 'Залишок',
}
